lib/resolvers/project.js

```diff
@@ -378,6 +378,12 @@ module.exports = ({ models }) => async ({ action, data, id, meta = {}, changedBy
 
     patch.raDate = getRaDate(versionToGrant, patch.expiryDate || project.expiryDate);
 
+    // activate any pending reminders created for the current version
+    await Reminder.query(transaction)
+      .patch({ status: 'active' })
+      .where({ modelId: project.id, status: 'pending' })
+      .where('createdAt', '>', versionToGrant.createdAt);
```

Comment:
What's the intention for the `createdAt` filter? There is possible scope for this to end up behaving weirdly if an inspector updates the conditions before returning to the applicant. That might be a really edgy edge case, but I can't see an obvious downside to removing the condition.

There's a possibility you might have old pending reminders from discarded amendments, so I think you'd need to ensure any pending reminders are deleted [when amendments are deleted](https://github.com/UKHomeOffice/asl-resolver/blob/master/lib/resolvers/project.js#L456-L478), but we probably want to do that anyway.

Reply:
This was to prevent any reminders left around from versions that did not get granted (i.e. discarded, rejected) accidentally being activated, but yeah deleting them is a better solution. Is `delete-amendments` triggered in both cases (discard / reject)?